android-components/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/OnDiskSitePermissionsStorage.kt

```diff
@@ -89,7 +89,7 @@ class OnDiskSitePermissionsStorage(
         includeTemporary: Boolean,
         private: Boolean,
     ): SitePermissions? {
-        if (private) return SitePermissions(origin, savedAt = 0)
+        if (private) return null
```

Comment:
Returning `null` is the right approach, as creating a new `SitePermissions` object intrinsically contains [default auto play values](https://github.com/mozilla-mobile/firefox-android/blob/a2dfa5952b7b31cd2aa5fe32699f01f8be33e0c1/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/permission/SitePermissions.kt#L26-L27), which removes the ability for apps to customize them. When permissions are not found in the storage, we [use the default actions given by apps](https://github.com/mozilla-mobile/firefox-android/blob/a2dfa5952b7b31cd2aa5fe32699f01f8be33e0c1/android-components/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt#L438).